src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -65,7 +65,24 @@ SyntaxNodeAnalysisContext ctx
 				return;
 			}
 
-			var unnamedArgs = GetUnnamedArgs(
+            // Expression trees aren't compatible with named arguments,
+            // so skip any expressions
+            var implicitType = ctx.SemanticModel.GetTypeInfo(ctx.Node.Parent).ConvertedType;
+            if (implicitType != null) {
+                // Only lambda type expressions have arguments,
+                // so this only applies to LambdaExpression
+                var expressionType = ctx.SemanticModel.Compilation.GetTypeByMetadataName("System.Linq.Expressions.LambdaExpression");

```

Comment:
Unfortunately we can't assume this exists. We'll have to check and if it doesn't, assume this isn't an expression tree.

Here's an example: https://github.com/Brightspace/D2L.CodeStyle/blob/c019e9c526f4d0c15fe83b3c1921aedb33d319ba/src/D2L.CodeStyle.Analyzers/ApiUsage/RpcAnalyzer.cs#L20-L32